### PR TITLE
Consolidate instrumentation documentation in docstrings

### PR DESCRIPTION
**Pull request by lzchen**
Updates instrumentation docstrings to include section on configuration. Most are related to `exclude_url` functionality and `request/response` hooks. We want all of our documentation to exist in one place (readthedocs).

### CODE REVIEW

**Comment by lzchen** [comment]
@codeboten 
I believe configuration options for instrumentations either exist in the `README` or in the docstrings of the instrumentation. We should choose one place and be consistent. Only choosing `README` in this PR because it seemed easier to find (from some feedback) but I don't have a strong opinion. Thoughts?

**Review by ocelotl — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-django/README.rst [hunk 1]
```diff
@@ -15,52 +15,6 @@ Installation
 
     pip install opentelemetry-instrumentation-django
```
Comment (codeboten) [inline instrumentation/opentelemetry-instrumentation-django/README.rst:17]: Where did this one go?

**Review by codeboten — COMMENTED** [review]
Just confirming this is moving all the documentation that would have ended up on pypi into the read the docs repo correct? Users should be able to access it via the links in each package.

File: instrumentation/opentelemetry-instrumentation-django/README.rst (hunk 1 above)
Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-django/README.rst:17]: It's already in [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/__init__.py#L14)

**Comment by lzchen** [comment]
@owais 
> We could also have our 4th generate script that allows us to keep docstrings as the single source of truth and generate everything else from that but this is fine as well as long as readthedocs contains the docs.

Haha so many generate scripts xD

**Comment by codeboten** [comment]
> @owais
> 
> > We could also have our 4th generate script that allows us to keep docstrings as the single source of truth and generate everything else from that but this is fine as well as long as readthedocs contains the docs.
> 
> Haha so many generate scripts xD

Soon we'll generate scripts via a generate script
5r7ki1
.

**Pull request merged**